components/dashboard/src/components/RepositoryFinder.tsx

```diff
@@ -0,0 +1,159 @@
+        if (JSON.stringify(results) !== JSON.stringify(searchResults)) {
+            setSearchResults(results);
+            setSelectedSearchResult(results[0]);
+        }
+    }
+
+    const search = async (query: string) => {
+        setSearchQuery(query);
+        await findResults(query, onResults);
+        if (await refreshSearchData(query, user)) {
+            // Re-run search if the underlying search data has changed
+            await findResults(query, onResults);
+        }
+    }
+
+    // Up/Down keyboard navigation between results
+    const onKeyDown = (event: React.KeyboardEvent) => {
+        if (!selectedSearchResult) {
+            return;
+        }
+        const selectedIndex = searchResults.indexOf(selectedSearchResult);
+        const select = (index: number) => {
+            // Implement a true modulus in order to "wrap around" (e.g. `select(-1)` should select the last result)
+            // Source: https://stackoverflow.com/a/4467559/3461173
+            const n = Math.min(searchResults.length, MAX_DISPLAYED_ITEMS);
+            setSelectedSearchResult(searchResults[((index % n) + n) % n]);
+        }
+        if (event.key === 'ArrowDown') {
+            event.preventDefault();
+            select(selectedIndex + 1);
+            return;
+        }
+        if (event.key === 'ArrowUp') {
+            event.preventDefault();
+            select(selectedIndex - 1);
+            return;
+        }
+    }
+
+    useEffect(() => {
+        const element = document.querySelector(`a[href='/#${selectedSearchResult}']`);
+        if (element) {
+            element.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+        }
+    }, [selectedSearchResult]);
+
+    const onSubmit = (event: React.FormEvent) => {
+        event.preventDefault();
+        if (selectedSearchResult) {
+            window.location.href = '/#' + selectedSearchResult;
+        }
+    }
+
+    return <form onSubmit={onSubmit}>
+        <div className="flex px-4 rounded-xl border border-gray-300 dark:border-gray-500">
+            <div className="py-4">
+                <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 16 16" width="16" height="16"><path fill="#A8A29E" d="M6 2a4 4 0 100 8 4 4 0 000-8zM0 6a6 6 0 1110.89 3.477l4.817 4.816a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 010 6z" /></svg>
+            </div>
+            <input type="search" className="flex-grow" placeholder="Repository" autoFocus value={searchQuery} onChange={e => search(e.target.value)} onKeyDown={onKeyDown} />
+        </div>
+        <div className="mt-3 flex flex-col space-y-2 h-64 overflow-y-auto" id="search-results">
+            {searchResults.slice(0, MAX_DISPLAYED_ITEMS).map((result, index) =>
+                <a className={`px-4 py-3 rounded-xl` + (result === selectedSearchResult ? ' bg-gray-600 text-gray-50 dark:bg-gray-700' : '')} href={`/#${result}`} key={`search-result-${index}`} onMouseEnter={() => setSelectedSearchResult(result)}>
```

Comment:
**issue**: There's a small vertical scroll nudge in some cases when hovering over the first or the last entry of the scrollbar area. Probably an after effect of having the entries wrapped? 🤷 

https://user-images.githubusercontent.com/120486/150562913-d9861788-57f5-473a-806e-9b0964abccd9.mov

 the scrollbar area. What do you think?

Reply:
That's because of the `scrollIntoView()` behavior when an element becomes selected (via keyboard or mouse selection). I didn't find this to be a problem when testing (and even found it helpful, e.g. if I hover over a partially hidden element, I can now read it fully).

Did you find this behavior surprising / unwelcome?

Reply:
I've seen both approaches work good, both scrolling or not scrolling when hovering or selecting a list entry inside a scrollable list. But I agree this small scroll nudge can be helpful. Let's leave it as is for now. 🤝